sips/sip-013/sip-013-semi-fungible-token-standard.md

```diff
@@ -0,0 +1,232 @@
+
+Semi-fungible tokens give operators the ability to create new token classes at will. They no longer need to deploy a new contract every time new token type is introduced. It greatly simplifies the flow for applications that require many new tokens and token types to come into existence.
+
+Benefits of using semi-fungible tokens:
+- Art NFTs can have series and be grouped in collections.
+- Games can have their in-game currencies and items easily represented.
+- DeFi protocols can leverage SFTs to transfer many tokens and settle multiple orders at once.
+- Easy bulk trades and transfers in a single contract call, saving on transaction fees.
+
+# Specification
+
+The Semi-Fungible Token trait, `sip013-semi-fungible-token-trait`, has 10 functions:
+
+## Trait functions
+
+### Balance
+
+`(get-balance ((token-id uint) (who principal)) (response uint uint))`
+
+Returns the token type balance `token-id` of a specific principal `who` as an unsigned integer wrapped in an `ok` response. It has to respond with `u0` if the principal does not have a balance of the specified token. The function should never return an `err` response and is recommended to be defined as read-only.
+
+### Overall balance
+
+`(get-overall-balance ((who principal)) (response uint uint))`
+
+Returns the overall SFT balance of a specific principal `who`. This is the sum of all the token type balances of that principal. The function has to respond with a zero value of `u0` if the principal does not have any balance. It should never return an `err` response and is recommended to be defined as read-only.
+
+### Total supply
+
+`(get-total-supply ((token-id uint)) (response uint uint))`
+
+Returns the total supply of a token type. If the token type has no supply or does not exist, the function should respond with `u0`. It should never return an `err` response and is recommended to be defined as read-only.
+
+### Overall supply
+
+`(get-overall-supply () (response uint uint))`
+
+Returns the overall supply of the SFT. This is the sum of all token type supplies. The function should never return an `err` response and is recommended to be defined as read-only.
+
+### Decimals
+
+`(get-decimals ((token-id uint)) (response uint uint))`
+
+Returns the decimal places of a token type. This is purely for display reasons, where external applications may read this value to provide a better user experience. The ability to specify decimals for a token type can be useful for applications that represent different kinds of assets using one SFT. For example, a game may have an in-game currency with two decimals and a fuel commodity expressed in litres with four decimals.
+
+### Token URI
+
+`(get-token-uri ((token-id uint)) (response (optional (string-utf8 256)) uint))`
```

Comment:
Why string-utf8 here? It can cause some problems, see e.g. https://www.w3.org/International/wiki/IRIStatus
I propose to follow sip-009 instead of sip-010.